test/functional/test_framework/test_node.py

```diff
@@ -741,7 +741,10 @@ def addconnection_callback(address, port):
             p2p_conn.wait_until(lambda: p2p_conn.message_count["version"] == 1, check_connected=False)
             p2p_conn.wait_until(lambda: not p2p_conn.is_connected, check_connected=False)
         else:
-            p2p_conn.wait_for_connect()
+            if reconnect:
+                p2p_conn.wait_for_reconnect()
```

Comment:
I think that this should be moved outside of the `if connection_type == "feeler":` clause and be done before:
After all the v2 reconnection logically stuff comes first, and then all the existing v1 logic still applies to the downgraded v1 connection - for example, the reconnection logic should also work if `add_outbound_p2p_connection` specified a feeler connection - in which case we currently wouldn't call `wait_for_reconnect`.

Reply:
true! done.